packages/core/useStorage/index.ts

```diff
@@ -218,7 +218,15 @@ export function useStorage<T extends(string | number | boolean | object | null)>
   }
 
   function update(event?: StorageEvent) {
-    if (event && event.key !== key)
+    if (event?.storageArea !== storage)
+      return
```

Comment:
This condition also fixes bug with using `useStorage` on `sessionStorage` and `localStorage` with the same key